### PR TITLE
[MSE] Remove MANAGED_MEDIA_SOURCE guards

### PR DESCRIPTION
**Pull request by philn**
#### 8cac4a32fbd675549de8ac52eed6b4d5c7125ac5
<pre>
[MSE] Remove MANAGED_MEDIA_SOURCE guards
<a href="https://bugs.webkit.org/show_bug.cgi?id=266174">https://bugs.webkit.org/show_bug.cgi?id=266174</a>

Reviewed by Jean-Yves Avenard and Xabier Rodriguez-Calvar.

The build with MANAGED_MEDIA_SOURCE disabled was unfixable. ManagedMediaSource is soon going to be
merged in the MSE spec, so there is no reason to keep those build guards. The runtime preference is
kept though.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::sourceType const):
* Source/WebCore/Modules/mediasource/BufferedChangeEvent.cpp:
* Source/WebCore/Modules/mediasource/BufferedChangeEvent.h:
* Source/WebCore/Modules/mediasource/BufferedChangeEvent.idl:
* Source/WebCore/Modules/mediasource/ManagedMediaSource.cpp:
* Source/WebCore/Modules/mediasource/ManagedMediaSource.h:
* Source/WebCore/Modules/mediasource/ManagedMediaSource.idl:
* Source/WebCore/Modules/mediasource/ManagedSourceBuffer.cpp:
* Source/WebCore/Modules/mediasource/ManagedSourceBuffer.h:
* Source/WebCore/Modules/mediasource/ManagedSourceBuffer.idl:
* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::addSourceBuffer):
(WebCore::MediaSource::memoryPressure):
* Source/WebCore/Modules/mediasource/MediaSource.h:
* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::sourceBufferPrivateBufferedChanged):
(WebCore::SourceBuffer::memoryPressure):
* Source/WebCore/Modules/mediasource/SourceBuffer.h:
* Source/WebCore/bindings/js/JSMediaSourceCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/bindings/js/JSSourceBufferCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/dom/EventNames.in:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::hasManagedMediaSource const):
(WebCore::HTMLMediaElement::mediaState const):
(WebCore::HTMLMediaElement::setBufferingPolicy):
(WebCore::HTMLMediaElement::purgeBufferedDataIfPossible):
* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Source/cmake/GStreamerDefinitions.cmake:
* Source/cmake/OptionsMac.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPageHasMediaStreamingActivity.mm:

Canonical link: <a href="https://commits.webkit.org/271975@main">https://commits.webkit.org/271975@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/4ffe41acebfd24137ac00acef373a5d4e89edd04

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29895 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8561 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31213 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32407 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27035 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30555 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10745 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5817 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27096 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30189 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7172 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25504 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6118 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6294 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26592 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33743 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/25679 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27268 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27017 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32465 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/30072 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6214 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4400 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30252 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7953 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/36464 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7159 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6958 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7871 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6741 "Built successfully") | | |

### CODE REVIEW

**Review by calvaris — APPROVED** [review]
lgtm but maybe you want to wait for @jyavenard 's blessing.

File: Source/WebCore/Modules/mediasource/SourceBuffer.cpp [hunk 1]
```diff
@@ -516,7 +516,11 @@ ExceptionOr<void> SourceBuffer::appendBufferInternal(const unsigned char* data,
     if (isRemoved() || m_updating)
         return Exception { ExceptionCode::InvalidStateError };
 
-    ALWAYS_LOG(LOGIDENTIFIER, "size = ", size, ", buffered = ", m_private->buffered(), " streaming = ", m_source->streaming());
+    ALWAYS_LOG(LOGIDENTIFIER, "size = ", size, ", buffered = ", m_private->buffered()
+#if ENABLE(MANAGED_MEDIA_SOURCE)
```
Comment (jyavenard) [inline Source/WebCore/Modules/mediasource/SourceBuffer.cpp]: isn't this the only thing preventing compilation if MMS is disabled?
why not limit to this single liner?

File: Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml [hunk 2]
```diff
@@ -4060,7 +4060,7 @@ ManagedMediaSourceEnabled:
   category: media
   humanReadableName: "Managed Media Source API"
   humanReadableDescription: "Managed Media Source API"
-  condition: ENABLE(MANAGED_MEDIA_SOURCE)
+  condition: ENABLE(MEDIA_SOURCE)
```
Comment (jyavenard) [inline Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml]: I don't see why this is needed

File: Source/WebKit/Shared/WebPreferencesDefaultValues.cpp [hunk 3]
```diff
@@ -175,10 +175,10 @@ bool defaultManageCaptureStatusBarInGPUProcessEnabled()
 
 #endif // ENABLE(MEDIA_STREAM)
 
-#if ENABLE(MANAGED_MEDIA_SOURCE) && ENABLE(MEDIA_SOURCE)
+#if ENABLE(MEDIA_SOURCE)
```
Comment (jyavenard) [inline Source/WebKit/Shared/WebPreferencesDefaultValues.cpp]: this seems broken, this method should only be enabled if MMS is compiled in

**Review by jyavenard — CHANGES_REQUESTED** [review]

**Comment by jyavenard** [comment]
I would remove the entire MANAGED_MEDIA_SOURCE as it will now be part of the official MSE spec. So if we build MEDIA_SOURCE we also build MANAGED_MEDIA_SOURCE

**Comment by philn** [comment]
I'll rework this then. Thanks for the feedback.

**Review by jyavenard — APPROVED** [review]

**Pull request merged**